lib/spack/spack/cmd/mirror.py

```diff
@@ -97,6 +98,19 @@ def setup_parser(subparser):
         default=spack.cmd.default_list_scope(),
         help="configuration scope to read from")
 
+    # Binary
+    upload_parser = sp.add_parser('binary', help=mirror_binary.__doc__)
+    upload_parser.add_argument('-d', '--directory', default=None,
+                               help="directory containing binaries for mirroring")
+    upload_parser.add_argument('-f', '--file', default=None,
+                               help="a file to upload to the mirror")
+    upload_parser.add_argument('-k', '--key', default=None,
```

Comment:
This should be implicit from the spec being uploaded to the mirror, so the user shouldn’t have to specify a key.